src/components/Products/ProductList.jsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { getAllProducts, deleteProduct } from "../../api/ProductService";
+import ProductForm from "./ProductForm";
+
+const ProductList = ({ setBasket }) => {
+  const [products, setProducts] = useState([]);
+  const [userRole, setUserRole] = useState("");
+
+  useEffect(() => {
+    loadProducts();
+    const token = localStorage.getItem('token');
+    const userRole = localStorage.getItem('user-role');
+    if (token) {
+      setUserRole(userRole);
+    }
+  }, []);
+
+  const loadProducts = async () => {
+    const products = await getAllProducts();
+    setProducts(products);
+  };
+
+  const handleDelete = async (id) => {
+    await deleteProduct(id);
+    loadProducts();
+  };
+
+  const addToBasket = (product) => {
+    setBasket((prev) => [...prev, product]);
+  };
+
+  return (
+    <div>
+      <h1 className="text-3xl font-bold mb-4">Products</h1>
+      {userRole === "admin" && <ProductForm onProductAdded={loadProducts} />}
+      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
+        {products.map((product) => (
+          <div key={product.id} className="p-4 bg-white shadow-md rounded-lg">
+            <img
+              src={`http://localhost:3000/${product.photo}`}
```

Comment:
create const of  BASE_URL in Axios interceptor, and import that here, don't use hardcoded urls